test/e2e/storage/drivers/base.go

```diff
@@ -74,6 +74,9 @@ type DynamicPVTestDriver interface {
 	// It will set fsType to the StorageClass, if TestDriver supports it.
 	// It will return nil, if the TestDriver doesn't support it.
 	GetDynamicProvisionStorageClass(fsType string) *storagev1.StorageClass
+
+	// GetClaimSize returns the size of the volume that is to be provisioned ("5Gi", "1Mi").
+	GetClaimSize() string
```

Comment:
Maximum size that a driver can return depends on the driver.
On the other hand, minimum size that a test requires depends on the test.

Please see [volumeIO test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volume_io.go#L304), [driver definition](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/base.go#L84), and [how it was arranged](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volume_io.go#L84). I prefer to do similar for it. (We could skip the test if a driver can't support the size that a test requires.)

If we are not going to add tests that the size matter in the future, it could be a over spec, though.
Also, if we would like to choose a short term solution, just wait for latest csi gce driver would be good, then we can change it back to 1Gi, which will fit to most drivers.

Reply:
So how about we let the driver define a minimum and maximum size for volumes that they can provision, and then tests can pick any size they like in that range when they provision volumes dynamically? Most tests don't care about the size at all and can just pick the minimum.

Reply:
Some drivers also have a min size

Reply:
@msau42 yes, that's why I suggested defining a range ("define a minimum and maximum size for volumes").

Reply:
@mkimuram while looking at how min/max size information could be used by a test, I ran into a snag:
https://github.com/kubernetes/kubernetes/blob/e9aabf97e1d969ed80719cb01d8acb4f2be221b7/test/e2e/storage/testsuites/base.go#L138-L169

That function must dynamically provision a volume, but it has no information about the size expected by the test because conceptually, that code is independent from any particular test.

Does it really make sense to specify a minimum and maximum size? I'm more leaning towards just letting the driver suggest a default size which may be larger than the absolute minimum supported by the storage system.

Optionally, tests can check whether the volume is large enough and skip otherwise. Which tests depend on a certain minimum size of the volume, and can they determine what the size of volume they operate against is? However, I would prefer to keep this out of this PR. Instead I would update the GetClaimSize comment explaining that the size must be chosen so that it is large enough for all tests that are intended to pass.

Reply:
@pohly with a single default size from the driver, how does the framework decide if a test requiring a minimum size should be run or not?

What if we had a field inside `TestConfig` to indicate the test's required size range, and also have a `GetClaimSizeRange` returned by the driver? `TestConfig` will then also get passed into `setupResources()`. Upon volume creation the framework always picks the minimum of the intersection between the test's range and the driver's range.

This way we don't artificially limit potential test <-> driver matches just because the default sizes don't work out

Reply:
I had tried to implement that, but quickly ran into problems because the code that uses the size doesn't always (or never? don't remember exactly) know which size the test that is going to run will need.

Can we leave solving that problem for the future? Taking the size from the driver definition is an improvement over hard-coding it in the tests, even if it isn't perfect.

Reply:
Done, see https://github.com/kubernetes/kubernetes/issues/72080